bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/hbasesnapshots/ImportJobFromHbaseSnapshot.java

```diff
@@ -0,0 +1,132 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.beam.hbasesnapshots;
+
+import com.google.bigtable.repackaged.com.google.api.core.InternalExtensionOnly;
+import com.google.cloud.bigtable.beam.CloudBigtableIO;
+import com.google.cloud.bigtable.beam.CloudBigtableTableConfiguration;
+import com.google.cloud.bigtable.beam.TemplateUtils;
+import com.google.cloud.bigtable.beam.sequencefiles.HBaseResultToMutationFn;
+import com.google.cloud.bigtable.beam.sequencefiles.ImportJob;
+import com.google.cloud.bigtable.beam.sequencefiles.Utils;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.io.hadoop.format.HadoopFormatIO;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PDone;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+
+/**
+ * A job that imports data from HBase snapshot exports hosted in Cloud Storage bucket into Cloud
+ * Bigtable. This job can be run directly or as a Dataflow template.
+ *
+ * <p>Execute the following command to run the job directly:
+ *
+ * <pre>
+ * mvn compile exec:java \
+ *   -DmainClass=com.google.cloud.bigtable.beam.hbasesnapshots.ImportJobFromHbaseSnapshot \
+ *   -Dexec.args="--runner=DataflowRunner \
+ *                --stagingLocation=gs://$STAGING_PATH \
+ *                --project=$PROJECT \
+ *                --bigtableInstanceId=$INSTANCE \
+ *                --bigtableTableId=$TABLE \
+ *                --gcsProject=$PROJECT \
+ *                --hbaseRootDir=gs://$HBASE_EXPORT_ROOT_PATH \
+ *                --snapshotName=$SNAPSHOT_NAME  \
+ *                --restoreDir=gs://$RESTORE_PATH
+ * </pre>
+ */
+@InternalExtensionOnly
+public class ImportJobFromHbaseSnapshot {
+  private static final Log LOG = LogFactory.getLog(ImportJobFromHbaseSnapshot.class);
+
+  public interface ImportOptions extends ImportJob.ImportOptions {
+    @Description("The GCP project id for the GCS bucket")
+    String getGcsProject();
+
+    @SuppressWarnings("unused")
+    void setGcsProject(String gcsProjectId);
+
+    @Description("The HBase root dir where HBase snapshot files resides.")
+    String getHbaseRootDir();
+
+    @SuppressWarnings("unused")
+    void setHbaseRootDir(String hbaseRootDir);
+
+    @Description("Temp location for restoring snapshots")
+    String getRestoreDir();
```

Comment:
Can you clarify what a temp location for restoring snapshots means? if its just a scratch dir, why not use a subdir in the job's temp directory?

Reply:
Added more document in the Class document

Reply:
why not use the beam's temp directory? why does a user need to specify a temp dir?

Reply:
Technically we could use a default temp dir, however, the restoring process will save some of the user data into the restoreDir, e.g, snapshot metadata, regioninfo, which could be sensitive.
Hence I figured we might want to make this explicit and let the users decide how to clean it up.
Otherwise I am ok to use `opts.getTempLocation()` for that path
What do you think?

Reply:
Take a look at how BigQueryIO handles it and see if its applicable:
https://github.com/xsm110/Apache-Beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java#L503-L533

Reply:
Created a DoFn to cleanup temp folders similar to https://github.com/apache/beam/blob/7c43ab6a8df9b23caa7321fddff9a032a71908f6/sdks/java/io/snowflake/src/main/java/org/apache/beam/sdk/io/snowflake/SnowflakeIO.java#L582 as it is simpler.

Had to use GcsUtil directly instead of beam FileSystem abstraction since the wrapping GcsFileSystem filterout directories intetionally while listing buckets.

PTAL